src/library_syscall.js

```diff
@@ -42,15 +42,7 @@ var SyscallsLibrary = {
     },
 
     doStat(func, path, buf) {
-      try {
-        var stat = func(path);
-      } catch (e) {
-        if (e && e.node && PATH.normalize(path) !== PATH.normalize(FS.getPath(e.node))) {
-          // an error occurred while trying to look up the path; we should just report ENOTDIR
-          return -{{{ cDefs.ENOTDIR }}};
-        }
-        throw e;
-      }
+      var stat = func(path);
```

Comment:
This looks like a significant functional change? Even with the change to `mayLookup` I can't figure out why this is NFC. Can you please elaborate?

Reply:
IIUC this change will not be observable to the callers of `stat`.   We have tests that cover this and all the tests pass so my claim is that this is just an internal refactor. 

AFAICT try/catch here was in place in order to rewrite the incorrect errno returned my mayLookup (EACCES) to the correct one (ENOTDIR).   My change to mayLookup fixes the errno the it returns and therefore obviates the need for this additional try/catch.

Reply:
For context this cleanup is part of a sequence of changes to cleanup and remove the errno usage in our JS library.  This is just the first step and it seems like right more to me.

Reply:
We do have stat tests, but the interesting logic `e.node && PATH.normalize(path) !== PATH.normalize(FS.getPath(e.node)` might not be tested. It was added here without a test:

6bac5f5832edb2834bc247a18de9ff16987db35f

for which I take full responsibility :smile: But, yeah,  I understand your explanation for `mayLookup` which sounds like it might handle this, and I verified `other.test_stat_fail_alongtheway` fails with only part of the patch, so looks like that's covered.

Reply:
Exactly yes. The `test_stat_fail_alongtheway` is the test that failed on the initial version of this PR.   I used it to figure out what was going on here and find this fix.